tracer/src/Datadog.Trace/Debugger/Sink/AgentBatchUploadApi.cs

```diff
@@ -53,5 +58,71 @@ public async Task<bool> SendBatchAsync(ArraySegment<byte> data)
 
             return true;
         }
+
+        private string BuildUri(string endpoint)
+        {
+            var uri = _apiRequestFactory.GetEndpoint(endpoint);
+            var builder = new UriBuilder(uri);
+            var query = HttpUtility.ParseQueryString(builder.Query);
+            _tags ??= GetDefaultTagsMergedWithGlobalTags();
+            query["ddtags"] = _tags;
+            builder.Query = query.ToString();
+            return builder.ToString();
+        }
+
+        private string GetDefaultTagsMergedWithGlobalTags()
+        {
+            var sb = StringBuilderCache.Acquire(StringBuilderCache.MaxBuilderSize);
+
+            try
+            {
+                var environment = TraceUtil.NormalizeTag(Tracer.Instance.Settings.EnvironmentInternal);
+                if (!string.IsNullOrEmpty(environment))
+                {
+                    sb.Append($"env:{environment},");
+                }
+
+                var version = Tracer.Instance.Settings.ServiceVersionInternal;
+                if (!string.IsNullOrEmpty(version))
+                {
+                    sb.Append($"version:{version},");
+                }
+
+                var hostName = PlatformHelpers.HostMetadata.Instance?.Hostname;
+                if (!string.IsNullOrEmpty(hostName))
+                {
+                    sb.Append($"host:{hostName},");
+                }
+
+                var runtimeId = Tracer.RuntimeId;
+                if (!string.IsNullOrEmpty(runtimeId))
+                {
+                    sb.Append($"{Tags.RuntimeId}:{runtimeId},");
+                }
+
+                var gitRepoUrl = Tracer.Instance.Settings.GitRepositoryUrl;
```

Comment:
This is not the correct way to get the repository url or the commit sha; this will only work if we obtained them through configuration, but we might be getting them through SourceLink.

Please use ` GitMetadataTagsProvider` (check out [example usages for more](https://github.com/DataDog/dd-trace-dotnet/blob/8f0857db0e71da254da1416a7b13d847643923a4/tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationManager.cs#L170-L190)).

Reply:
Fixed in 0e66cde62e86d91727c438704c328b85fc7700a1